backend/app/models/uni_models.py

```diff
@@ -118,105 +126,94 @@ class Config:
-                "startHour": "8.10",
-                "dLesson": 50,
-                "dBreak": 10,
-                "slotCount": 12,
-            }
+            "example": {"_id": "61fc266ae3d749b1d65c17c6", "name": "20-21 Spring",}
         }
 
 
 class CurriculumLessonModel(BaseModel):
     id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
-    name: str = Field(...)
-    code: str = Field(...)
-    lessonType: str = Field(...)
+    name: str = Field(..., min_length=1, max_length=100)
+    code: str = Field(..., min_length=1, max_length=100)
+    lesson_type: str = Field(..., min_length=1, max_length=30)
 
     class Config:
         allow_population_by_field_name = True
         arbitrary_types_allowed = True
         json_encoders = {ObjectId: str}
         schema_extra = {
-            "example": {"name": "PHYSICS I", "code": "PHYS101", "lessonType": "science"}
+            "example": {
+                "name": "PHYSICS I",
+                "code": "PHYS101",
+                "lesson_type": "science",
+            }
         }
 
 
 class CurriculumSemesterModel(BaseModel):
     id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
-    semester: int = Field(..., gt=0, lt=9)
+    number: int = Field(..., gt=0, lt=20)
     lessons: List[CurriculumLessonModel] = []
 
     class Config:
         allow_population_by_field_name = True
         arbitrary_types_allowed = True
         json_encoders = {ObjectId: str}
-        schema_extra = {
-            "example": {
-                "semester": 1,
-            }
-        }
+        schema_extra = {"example": {"number": 1,}}
 
 
 class UniversityCurriculumModel(BaseModel):
     id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
     name: str = Field(..., min_length=1)
-    startYear: int = Field(...)
-    endYear: int = Field(...)
+    start_year: int = Field(...)
+    end_year: int = Field(...)
     semesters: List[CurriculumSemesterModel] = []
 
     class Config:
         allow_population_by_field_name = True
         arbitrary_types_allowed = True
         json_encoders = {ObjectId: str}
         schema_extra = {
-            "example": {"name": "2016 Later", "startYear": 2016, "endYear": 2100}
+            "example": {"name": "2016 Later", "start_year": 2016, "end_year": 2100}
         }
 
 
 class UniversityDepartmentModel(BaseModel):
     id: PyObjectId = Field(default_factory=PyObjectId, alias="_id")
-    name: str = Field(..., min_length=2 ,max_length=100,)  
+    name: str = Field(..., min_length=2, max_length=100)
     curriculums: List[UniversityCurriculumModel] = []
 
     class Config:
         allow_population_by_field_name = True
         arbitrary_types_allowed = True
         json_encoders = {ObjectId: str}
-        schema_extra = {
-            "example": {
-                "name": "COMP",
-            }
-        }
+        schema_extra = {"example": {"name": "COMP",}}
 
 
 class UniversityModel(BaseModel):
```

Comment:
We may need to left some uni_model fields Optional.

Reply:
If we want them to be optional, we should find a way for MongoDB to update few of them or we can iterate through the attributes of the model and if it is not none we can update it. Which one will be better?

Reply:
Same with upper case.

Reply:
Same solution for above can be used here.